### PR TITLE
Protobuf: bump to 3.15.8

### PR DESCRIPTION
**Pull request by ktf**
Needed by Arrow 3.0 on some platforms.

### CODE REVIEW

**Review by teo — APPROVED** [review]

**Review by teo — APPROVED** [review]

**Comment by ktf** [comment]
@teo looks like this is creating troubles to `occ.pb.cc`, can you have a look?

**Comment by ktf** [comment]
@ironMann can you take a look at why DD fails with the new version of protobuf? For a similar issue in AliECS @teo had to force C++14 compilation, IIUC.

**Comment by ironMann** [comment]
@ktf it's prob this: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=82461
Seems to be solved as of gcc 7.4

Stepping down from c++17 is unfortunately not an option now for DD. There are also other protobuf users, dds, odc, ...

**Comment by davidrohr** [comment]
Sounds like yet another reason to go to GCC 10.2 :)

**Comment by ktf** [comment]
@davidrohr Yeah... As long as we do not start using C++20, since that will break ROOT...

**Comment by ktf** [comment]
Ok I've bumped #2668 to use GCC 10.2. Seems to work fine on my ubuntu box, including the autotools-archive stuff...`--std=c++2y` here we come! ;-)

**Comment by ktf** [comment]
@vascobarroso @awegrzyn could you comment if it's really required for FLP to stick to the system compiler for cc7? It will require us to do some juggling to get the new arrow in, if that is the case.

**Review by vascobarroso — DISMISSED** [review]

**Comment by teo** [comment]
@ktf can we merge this? We could use it now for the Suite beta release.

File: defaults-o2-dataflow.sh [hunk 1]
```diff
@@ -30,6 +30,8 @@ disable:
   - O2sim
   - O2-full-system-test
 overrides:
+  protobuf:
```
Comment (awegrzyn) [inline defaults-o2-dataflow.sh:33]: @teo I think this needs to be removed before?

Comment (teo) [reply defaults-o2-dataflow.sh:33]: Sounds right, is there any impediment to doing so?

Comment (vascobarroso) [reply defaults-o2-dataflow.sh:33]: @ktf could you confirm that we can go ahead with this suggestion ?

Comment (ktf) [reply defaults-o2-dataflow.sh:33]: Not until you are happy with dropping GCC 7.3, since 5.15.8 will not compile there.

Comment (ktf) [reply defaults-o2-dataflow.sh:33]: Alternative @ironMann should find a workaround like you did to use 3.15.8 in DD & the rest of the EPN stack.

Comment (teo) [reply defaults-o2-dataflow.sh:33]: I will unbump the minimum version of Protobuf for OCC back to v3.14.0 so we can still build the FLP Suite with defaults `o2-dataflow`.

Comment (teo) [reply defaults-o2-dataflow.sh:33]: Fix here https://github.com/alisw/alidist/pull/3014

**Review by vascobarroso — APPROVED** [review]

**Comment by vascobarroso** [comment]
@alisw/infrastructure-admins this looks green now. Could it be merged, we need this for the release of FLP Suite v0.18.0 beta.

**Pull request merged**